core/src/main/scala/org/apache/spark/sql/execution/row/RowFormatScanRDD.scala

```diff
@@ -411,7 +414,7 @@ final class InternalRowIteratorOnRS(conn: Connection,
       i = i + 1
     }
     moveNext()
-    mutableRow
+    unsafeproj.apply(mutableRow)
```

Comment:
Yes, codegeneration is the way to go but I thought that directly creating an UnsafeRow would be equally slow.